wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,343 @@
+from discord.ext import commands
+import discord
+from resources.utilities.embed import embed as em
+from WalleModels.models import BannedUsers, BanRecords
+import datetime
+import pytz
+import re
+from typing import Union
+import logging
+from asgiref.sync import sync_to_async
+logger = logging.getLogger('wall_e')
+
+
+class Ban(commands.Cog):
+
+    def __init__(self, bot, config):
+        self.bot = bot
+        self.config = config
+        self.blacklist = []
+        self.errorColour = 0xA6192E
+
+    @sync_to_async
+    def insert_ban(self, username, user_id):
+        """Adds entry to BannedUsers table"""
+
+        logger.info(f"[Ban insert_ban()] Adding banned user to Banned_user with values ({username, user_id})")
+        BannedUsers(username, str(user_id)).save()
+
+    @sync_to_async
+    def insert_record(self, username, user_id, mod, mod_id, date, reason):
+        """Adds entry to BanRecords table"""
+
+        logger.info("[Ban insert_record()] Adding ban record to BanRecords with values" +
+                    f"({username, user_id, mod, mod_id, date, reason})")
+
+        BanRecords(username=username, user_id=str(user_id), mod=mod, mod_id=str(mod_id), date=date,
+                   reason=reason).save()
+
+    @sync_to_async
+    def get_banned_ids(self):
+        """Gets list of all user_ids from BannedUsers table"""
+
+        return list(map(int, (BannedUsers.objects.values_list('user_id', flat=True))))
+
+    @commands.Cog.listener(name='on_ready')
+    async def load(self):
+        """Grabs channel to send mod reports to and reads in the blacklist from db"""
+
+        logger.info('[Ban load()] Attempting to get the #council-summary channel')
+        self.mod_channel = discord.utils.get(self.bot.guilds[0].channels, name="council-summary")
```

Comment:
please update this so that `council-summary` is defined in the `ini files instead